src/components/program-settings.js

```diff
@@ -315,103 +367,126 @@ class ProgramSettings extends React.Component {
         this.setState({
             specificSetting: {
```

Comment:
I'm not sure I understand what the difference is between `specificSetting.teiDownload` and `specificteiDownload` - why do we have both?  Also, we should always use proper Camel Case for javascript variables, so it should be `specificTeiDownload` or `specificTEIDownload` (as it was before)

This applies to all of these settings.

Note that React auto-merges the state objects, but only at the top level, so the extra level `state.specificSetting` needs to be manually merged if you're using it.

Reply:
this was fixed in fix/duplicatedStates branch